src/supervisor/watchers/index.ts

```diff
@@ -60,9 +60,18 @@ export async function beginWatchingWorkloads(): Promise<void> {
       { namespace: config.WATCH_NAMESPACE },
       'kubernetes-monitor restricted to specific namespace',
     );
-    const namespaceResponse = await k8sApi.coreClient.readNamespace(
-      config.WATCH_NAMESPACE,
-    );
+    let namespaceResponse: NamespaceResponse | undefined;
+    try {
+      namespaceResponse = await k8sApi.coreClient.readNamespace(
+        config.WATCH_NAMESPACE,
+      );
+    } catch (err) {
+      logger.error(
+        { watchedNamespace: config.WATCH_NAMESPACE, err },
+        'failed to read the namespace',
+      );
+      return;
```

Comment:
Just to note that this will just end the application so it will keep restarting in a loop 😄

Reply:
I think this is ok for the for running in 1 namespace? should we retry?

Reply:
but now that we give the permission, it shouldn't return err. If they did return err, we want to log it and crash the app, so they will come to us with this error quicker?